sklearn/base.py

```diff
@@ -292,6 +293,20 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
                                                offset=len(class_name),),)
 
+    def __getstate__(self):
+        return dict(self.__dict__.items(), __version__=__version__)
+
+    def __setstate__(self, state):
+        pickle_version = state.get("__version__", "pre-0.18")
+        if pickle_version != __version__:
+            warnings.warn(
+                "Trying to unpickle estimator from version {} when with using "
+                "version {}. This might lead to breaking code or invalid "
+                " results. Use at your own risk.".format(
+                    pickle_version, __version__), UserWarning)
```

Comment:
I think we might want to give this a custom warning type so that someone can easily silence it. (Then again, silencing by message regexp might be safer??)